### PR TITLE
sentry events

### PR DESCRIPTION
**Pull request by peterbe**
Part of #1 

It works! See https://sentry.prod.mozaws.net/operations/rs-uptake-health-dev/issues/5283976/?query=is:unresolved

I suspect, that the way Sentry groups events is by lumping it together by the whole `message` passed into `sentry_sdk.capture_message()`. And possibly the scope too. 

Either way, first I thought I'd make the message something like:

```python
f"{source} is erroring too much on {today.strftime('%Y-%m-%d')}\n"
```
so that it'd be different every day. But if don't do that, what determines if it's a new error in Sentry is if the numbers are any different. And they won't be day-to-day.

### CODE REVIEW

**Comment by peterbe** [comment]
So, I stumbled on a nasty pip bug. See: https://github.com/pypa/pip/issues/4780#issuecomment-467185924

Basically, because I changed the `name` in `setup.py`, now if you do `pip install -e ".[dev,test]"` it won't pick up any of the new changes I added to `install_requires`. To remedy that, delete any `*.egg-info` directories.

File: main.py [hunk 1]
```diff
@@ -217,11 +222,33 @@ def is_neutral(status):
...
+    use_sentry = False
+    if SENTRY_DSN:
+        sentry_sdk.init(SENTRY_DSN)
+        use_sentry = True
+    elif not DEBUG:
+        click.secho("No SENTRY_DSN set but not in DEBUG mode!", bold=True)
+    try:
+        bads = run(verbose=verbose, dry_run=dry_run)
+    except Exception as exception:
+        sentry_sdk.capture_exception(exception)
```
Comment (leplatrem) [inline main.py]: It was not obvious on first read why we had `capture_exception()` and `capture_message()`.
Could you please add a comment that here we capture «unexpected» errors like network, parsing or whatever. And that we report bad status below?

File: main.py [hunk 2]
```diff
@@ -233,7 +260,10 @@ def cli(dry_run, verbose):
             )
             click.secho(f"\n{source}\n{stats}")
 
-        raise click.Abort
+            if use_sentry:
```
Comment (leplatrem) [inline main.py]: nit: is it really necessary to have this `use_sentry` variable?  `if SENTRY_DSN` wouldn't be too bad IMO ;)

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 *.toml
 remote_settings_uptake_worrying.egg-info
```
Comment (leplatrem) [inline .gitignore]: we could probably remove this I think...

Comment (peterbe) [reply .gitignore]: And the `*.toml` line above :)

File: main.py (hunk 2 above)
Comment (peterbe) [reply main.py]: Removed. 

I've tested it. If you don't have a `SENTRY_DSN` and thus never called `sentry_sdk.init()` all future calls to `sentry_sdk.capture_exception` or `sentry_sdk.capture_message` are just no-ops that do nothing and breaks nothing.

File: main.py (hunk 1 above)
Comment (peterbe) [reply main.py]: I believe I have.

**Pull request merged**